### ISSUE TITLE
Fix SonarQube distinctions between production and development dependencies

### ISSUE THREAD

**Issue by okorach** [okorach/sonar-tools#2283]
When analysing `sonar-tools` with SonarQube Advanced Security, `pylint` is incorrectly reported as a `production` dependency instead of a `development` one.
The project configuration should be changed so that deps are properly classified

<img width="1297" height="805" alt="Image" src="https://github.com/user-attachments/assets/7a5f6a34-3cb0-4897-b2ef-8e1c8559885c" />

**Comment by okorach** [comment]
Hints to help the fix:
```
In Poetry, dependency separation is handled through dependency groups, defined in pyproject.toml.
Modern approach (Poetry ≥1.2): [tool.poetry.group.*]

toml[tool.poetry.dependencies]
python = "^3.11"
requests = "^2.31.0"
fastapi = "^0.110.0"

[tool.poetry.group.dev.dependencies]
pytest = "^8.0.0"
black = "^24.0.0"
mypy = "^1.9.0"

[tool.poetry.group.test.dependencies]
pytest = "^8.0.0"
pytest-cov = "^5.0.0"

[tool.poetry.group.build.dependencies]
setuptools = "^69.0.0"
wheel = "^0.43.0"
Key points:

Anything under [tool.poetry.dependencies] is a production/runtime dependency — it ships with your package and is required to actually run the code.
Anything under [tool.poetry.group.<name>.dependencies] is scoped to that named group (dev, test, build, docs, etc. — you can name groups whatever you want).
Groups are not installed by default when someone installs your package from PyPI; they only matter in the local dev workflow via poetry install.

Installing/excluding groups
bashpoetry install                       # installs main deps + all non-optional groups
poetry install --only main           # production deps only
poetry install --without dev,test    # skip dev and test groups
poetry install --with docs           # include an optional group
Optional groups
You can mark a group as optional so it's excluded unless explicitly requested:
toml[tool.poetry.group.docs]
optional = true

[tool.poetry.group.docs.dependencies]
mkdocs = "^1.5.0"
Then it's only pulled in with poetry install --with docs.
Legacy syntax (pre-1.2, still seen in older projects)
Older projects used [tool.poetry.dev-dependencies] for anything non-production. This still works for backward compatibility but is deprecated in favor of group.dev.
Why this matters for production builds
When building a production artifact (Docker image, deployment package), you typically run:
bashpoetry install --only main --no-root
This ensures test/lint/build tooling never ends up in your production image, keeping it smaller and reducing the attack surface from unnecessary packages.
Quick mental model
SectionPurposeShipped to end users?[tool.poetry.dependencies]Runtime/production depsYes[tool.poetry.group.dev.dependencies]Dev tooling (linters, formatters)No[tool.poetry.group.test.dependencies]Test frameworks/pluginsNo[tool.poetry.group.build.dependencies]Build-time toolsNo
One nuance: if you're publishing a library (not just deploying an app), only [tool.poetry.dependencies] (plus any extras) ends up in the published wheel's metadata — group dependencies are purely a local/CI convenience and never appear in the built package's dependency list at all.
```
